soc/arc/snps_qemu/tune_build_ops.cmake

```diff
@@ -0,0 +1,2 @@
+# workaround for issue #54720
+list(APPEND TOOLCHAIN_C_FLAGS -fno-optimize-sibling-calls)
```

Comment:
Let's just disable the `tests/kernel/mem_protect/syscalls` test for `qemu_arc_em`. It's the QEMU bug and not a code generation issue. By changing the code we may simply workaround the issue for this test case but it may appear for another.

Reply:
I tend to disagree. Removing tail calls is most likely to work all possible issues with QEMU for all tests. Disabling this one test makes it more likely that someone's unrelated change will trigger this QEMU bug in another test (just happened tome). So please let's go with this wide ranging workaround for the release please.

Reply:
@evgeniy-paltsev Given that this is only specified for `snps_qemu`, which is a QEMU-only target, I think this temporary workaround is reasonable (it would be a different story if this flag were specified globally for all ARC platforms; but, that is not the case).

Reply:
Agree that this matches the working theory best.  Disadvantage is that we're testing ever so slightly different code in CI, but the advantage is we don't lose coverage on the case that's faulting (FWIW: I bet anything we're seeing low-frequency faults elsewhere, it's just mem_protect/syscalls was able to fail deterministically and get through the retries).

Reply:
All, please note that the instruction sequence triggering the exception is NOT an illegal sequence. Root cause analysis has not been concluded, it is a tricky issue. But everything points to a Qemu issue, so fixing it by modifying timing/behavior through compiler options or other ways (Use `-singlestep`; Increase/decrease the CPU frequency: `-global cpu.freq_hz=100000000` (100 MHz)) I consider 'lucky' workarounds for now.
I tend to agree with Evgeniy best is to disable this test for now. There's enough coverage of mem_protect/syscalls on other ARC targets.

Reply:
@ruuddw @evgeniy-paltsev In that case, can you please send a pull request to disable the test?

Reply:
> @ruuddw @evgeniy-paltsev In that case, can you please send a pull request to disable the test?

Done, see #54910